### PR TITLE
Add documentation for handleSignUp

### PR DESCRIPTION
**Pull request by wlee221**
_Issue #, if available:_ closes https://github.com/aws-amplify/amplify-js/issues/8275

_Description of changes:_ Adds documentation for `handleSignUp` added in https://github.com/aws-amplify/amplify-js/issues/8474.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/ui/auth/fragments/web/authenticator.md [hunk 1]
```diff
@@ -1068,6 +1068,35 @@ To achieve this, you first need to setup the userpool to allow email or phone nu
 
 </docs-filter>
 
+### Override Sign Up
+
+You can override the call to [`signUp`](~/lib/auth/emailpassword.md) function with `handleSignUp` property. For example, you can add a [custom attribute](~/lib/auth/emailpassword.md) like so:
```
Comment (wlee221) [inline docs/ui/auth/fragments/web/authenticator.md:1073]: I tried using 
```
[custom attribute](~/lib/auth/emailpassword.md#custom-attributes)
```
to link to exact subsection, but the page wasn't navigating to the correct subsection so I just linked it to `~/lib/auth/emailpassword.md` for now.

Comment (wlee221) [reply docs/ui/auth/fragments/web/authenticator.md:1073]: Kapture 2021-06-24 at 17 41 46
Maybe this is a bug or I'm missing something? cc @tjleing

**Review by eddiekeller — APPROVED** [review]
LGTM! Thanks for updating the docs!

File: docs/ui/auth/fragments/web/authenticator.md (hunk 1 above)
Comment (wlee221) [reply docs/ui/auth/fragments/web/authenticator.md:1073]: Talked offline, found that
- opening a new tab works
- looks like the docs is trying to scroll to the original height before navigation

and will be investigated.

This is not blocking this PR, so we'll go ahead and get this reviewed / merged.

File: docs/ui/auth/fragments/vue/overriding-sign-up-submit.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	</div>
+</template>
+<script>
+import { Auth } from '@aws-amplify/auth';
+
+export default {
+	name: 'Authenticator',
+	methods: {
+		handleSignUp(data) {
+			console.log(data);
```
Comment (sammartinez) [inline docs/ui/auth/fragments/vue/overriding-sign-up-submit.md:22]: Why would want to `console.log()` this data?

Comment (wlee221) [reply docs/ui/auth/fragments/vue/overriding-sign-up-submit.md:22]: Nice catch, adjusting

Comment (wlee221) [reply docs/ui/auth/fragments/vue/overriding-sign-up-submit.md]: Found some other typos too so I pushed https://github.com/aws-amplify/docs/pull/3346/commits/70719cee6a891cc73b7f3332ce9732c620672d44 to handle those! The infamous @wlee221 clumsiness 😄

File: docs/ui/auth/fragments/vue/overriding-sign-up-submit.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        ></amplify-sign-up>
+    </amplify-authenticator>
+</template>
+<script>
+import { Auth } from 'aws-amplify';
+
+export default {
+	name: 'Authenticator',
+	methods: {
+		handleSignUp(data) {
```
Comment (sammartinez) [inline docs/ui/auth/fragments/vue/overriding-sign-up-submit.md:18]: Where is `data` used?

**Review by sammartinez — COMMENTED** [review]
One comment and I can approve the PR

File: docs/ui/auth/fragments/vue/overriding-sign-up-submit.md (hunk 3 above)
Comment (wlee221) [reply docs/ui/auth/fragments/vue/overriding-sign-up-submit.md:18]: Ah, this should be `formData`! Thanks for the catch.
```suggestion
		handleSignUp(formData) {
```

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**